File: spring-project/src/main/java/ktc/spring_project/entities/Address.java
package ktc.spring_project.entities;

import jakarta.persistence.*;
import ktc.spring_project.enums.AddressType;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.sql.Timestamp;

@Entity
@Table(name = "addresses")
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @Enumerated(EnumType.STRING)
@Column(name = "address_type", nullable = false, length = 50)
private AddressType addressType;


    @Column(nullable = false, length = 500)
    private String address;

    @Column(precision = 10, scale = 8)
    private BigDecimal latitude;

    @Column(precision = 11, scale = 8)
    private BigDecimal longitude;

    @Column(length = 100)
    private String city;

    @Column(length = 100)
    private String state;

    @Column(length = 100)
    private String country = "Vietnam";

    @Column(length = 100)
    private String region;

    @Column(name = "postal_code", length = 20)
    private String postalCode;

    @Column(name = "contact_name", length = 255)
    private String contactName;

    @Column(name = "contact_phone", length = 20)
    private String contactPhone;

    @Column(name = "contact_email", length = 255)
    private String contactEmail;

    @Column(name = "floor_number", length = 10)
    private String floorNumber;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
  
public AddressType getAddressType() {
    return addressType;
}

public void setAddressType(AddressType addressType) {
    this.addressType = addressType;
}

    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
    public BigDecimal getLatitude() { return latitude; }
    public void setLatitude(BigDecimal latitude) { this.latitude = latitude; }
    public BigDecimal getLongitude() { return longitude; }
    public void setLongitude(BigDecimal longitude) { this.longitude = longitude; }

    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }
    public String getState() { return state; }
    public void setState(String state) { this.state = state; }
    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }
    public String getRegion() { return region; }
    public void setRegion(String region) { this.region = region; }
    public String getPostalCode() { return postalCode; }
    public void setPostalCode(String postalCode) { this.postalCode = postalCode; }
    public String getContactName() { return contactName; }
    public void setContactName(String contactName) { this.contactName = contactName; }
    public String getContactPhone() { return contactPhone; }
    public void setContactPhone(String contactPhone) { this.contactPhone = contactPhone; }
    public String getContactEmail() { return contactEmail; }
    public void setContactEmail(String contactEmail) { this.contactEmail = contactEmail; }
    public String getFloorNumber() { return floorNumber; }
    public void setFloorNumber(String floorNumber) { this.floorNumber = floorNumber; }
    public Timestamp getCreatedAt() { return createdAt; }
    public void setCreatedAt(Timestamp createdAt) { this.createdAt = createdAt; }
    public Timestamp getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Timestamp updatedAt) { this.updatedAt = updatedAt; }
}

File: flutter-project/lib/domain/models/order/address.dart
// address.dart
// Model cho thông tin địa chỉ

class Address {
  final int id;
  final String? addressType;
  final String address;
  final double? latitude;
  final double? longitude;
  final String? city;
  final String? state;
  final String? country;
  final String? postalCode;
  final String? contactName;
  final String? contactPhone;
  final String? contactEmail;
  
  Address({
    required this.id,
    this.addressType,
    required this.address,
    this.latitude,
    this.longitude,
    this.city,
    this.state,
    this.country,
    this.postalCode,
    this.contactName,
    this.contactPhone,
    this.contactEmail,
  });
  
  factory Address.fromJson(Map<String, dynamic> json) => Address(
    id: json["id"] ?? 0,
    addressType: json["addressType"],
    address: json["address"] ?? json["fullAddress"] ?? "No address",
    latitude: json["latitude"] != null 
        ? double.tryParse(json["latitude"].toString()) 
        : null,
    longitude: json["longitude"] != null 
        ? double.tryParse(json["longitude"].toString()) 
        : null,
    city: json["city"],
    state: json["state"],
    country: json["country"],
    postalCode: json["postalCode"],
    contactName: json["contactName"],
    contactPhone: json["contactPhone"],
    contactEmail: json["contactEmail"],
  );
  
  Map<String, dynamic> toJson() => {
    "id": id,
    "addressType": addressType,
    "address": address,
    "latitude": latitude,
    "longitude": longitude,
    "city": city,
    "state": state,
    "country": country,
    "postalCode": postalCode,
    "contactName": contactName,
    "contactPhone": contactPhone,
    "contactEmail": contactEmail,
  };
}
